plugins/modules/datahub_cluster.py

```diff
@@ -549,10 +553,11 @@ def main():
             delay=dict(required=False, type='int', aliases=['polling_delay'], default=15),
             timeout=dict(required=False, type='int', aliases=['polling_timeout'], default=3600)
         ),
-        supports_check_mode=True,
-        required_together=[
-            ['subnet', 'image', 'catalog', 'template', 'groups', 'environment'],
-        ]
+        supports_check_mode=True
+        #Punting on additional checks here. There are a variety of supporting datahub invocations that can make this more complex
```

Comment:
I'd like to tackle this logic, as these checks allow us to "fail fast" when using the module.

Reply:
I'll defer to someone smarter than me here. A.) I don't know how the Ansible logic works and B.) I don't actually know what CDP requires here. For example, I don't think this was ever actually correct. You don't NEED a subnet. What I do know is, that when you provide a cluster definition, all you need is an environment name, cluster definition, and datahub name. 

I'm not really in the position to go cloud by cloud and figure out all the permutations :) - is catalog required? Probably not.

Reply:
I have reviewed the currently implemented checks, and they are in fact not strictly correct but do pass our test cases.
I suggest it might be helpful to raise an Issue that the input checks in datahub_cluster need revision and leave them out here?